File: backend/src/NobetSistemi.Domain/Entities/LeaveRequest.cs
using NobetSistemi.Domain.Enums;

namespace NobetSistemi.Domain.Entities;

public class LeaveRequest
{
    public Guid Id { get; set; }
    public Guid GroupId { get; set; }
    public Group Group { get; set; } = null!;
    public Guid UserId { get; set; }
    public User User { get; set; } = null!;
    public DateTime StartDate { get; set; }
    public DateTime EndDate { get; set; }
    public string Reason { get; set; } = string.Empty;
    public LeaveStatus Status { get; set; } = LeaveStatus.Pending;
    public Guid? ReviewedById { get; set; }
    public User? ReviewedBy { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}

File: frontend/nobet_app/lib/models/leave_request.dart
class LeaveRequest {
  final String id;
  final String userId;
  final String userName;
  final String startDate;
  final String endDate;
  final String reason;
  final String status;
  final String? reviewedById;
  final String createdAt;

  const LeaveRequest({
    required this.id,
    required this.userId,
    required this.userName,
    required this.startDate,
    required this.endDate,
    required this.reason,
    required this.status,
    this.reviewedById,
    required this.createdAt,
  });

  factory LeaveRequest.fromJson(Map<String, dynamic> j) => LeaveRequest(
        id: j['id'],
        userId: j['userId'],
        userName: j['userName'] ?? '',
        startDate: j['startDate'],
        endDate: j['endDate'],
        reason: j['reason'] ?? '',
        status: j['status'] ?? 'Pending',
        reviewedById: j['reviewedById'],
        createdAt: j['createdAt'] ?? '',
      );
}
